utils/examples/generate-examples.py

```diff
@@ -0,0 +1,108 @@
+   'outputFormatRaw': True,
+   'removeIds': True,
+   'svgFormatRaw': True,
+   'svgHtml5': True,
+   'svgRemoveXlink': True,
+   'svgViewBox': True,
+    'xmlIdSeed': 1
+}
+
+success: bool = True
+
+if __name__ == "__main__":
+    description = """
+        Renders SVG for the MEI examples. Will generate any new examples found; to
+        re-generate all examples use the '--clean' option.
+    """
+    parser = argparse.ArgumentParser(description=description)
+    parser.add_argument("--clean", action="store_true", help="Re-generate all examples.")
+    verbose_group = parser.add_mutually_exclusive_group()
+    verbose_group.add_argument("--verbose", "-v", action="store_true")
+    verbose_group.add_argument("--debug", "-d", action="store_true")
+
+    args = parser.parse_args()
+    if args.debug:
+        level_msg: str = "Debug"
+        level = logging.DEBUG
+    elif args.verbose:
+        level_msg: str = "Info"
+        level = logging.INFO
+    else:
+        level_msg: str = "Error"
+        level = logging.ERROR
+
+    logging.basicConfig(format="[%(asctime)s] [%(levelname)8s] %(message)s (%(filename)s:%(lineno)s)",
+                        level=level)
+    log = logging.getLogger(__name__)
+    log.info("Running at logging level %s", level_msg)
+
+    verovio.enableLog(verovio.LOG_DEBUG)
+    tk = verovio.toolkit()
+    # version of the toolkit
+    log.info("Using Verovio %s", tk.getVersion())
+
+    # keep all the options to be able to reset them for each example
+    default_options: dict = tk.getDefaultOptions()
+    # Overwrite the default options with our locally-defined options
+    default_options.update(VRV_OPTIONS)
+
+    for file in os.listdir(IMAGES_PATH):
+        # Skip everything that is not an .mei or .xml file
+        if not(file.endswith(".mei")) and not(file.endswith(".xml")): 
+            continue
+        
+        options: dict = default_options.copy()
+
+        mei_file = os.path.join(IMAGES_PATH, file)
+        svg_file = os.path.join(IMAGES_PATH, f"{file}.svg")
+
+        if os.path.exists(svg_file) and os.path.exists(mei_file) and not args.clean:
+            log.info("This example already exists, and we're not running in cleaning mode. Skipping.")
+            continue
+
+        # Download the MEI file from the given url
+        mei_example: str = ""
+        with open(mei_file) as f:
+            mei_example = f.read()
+
+        log.debug("Running Verovio with the following options: %s", pprint.pformat(options))
+        tk.setOptions(options)
+        if not(tk.loadData(mei_example)):
+            log.error("Failed to load %s", mei_file)
+            success = False
+            continue
+
+        svg: str = tk.renderToSVG(1)
+
+        if len(svg):
+            with open(svg_file, 'w') as f:
+                f.write(svg)
+                log.info("Successfully rendered %s", svg_file)
+        else:
+            log.error("Failed to render %s", mei_file)
+            success = False
```

Comment:
```suggestion
            success &= False
```

The reason is that you're constantly resetting the `success` variable to the outcome of the last run, rather than tracking whether all renders were executed successfully. `True & True & False & True = False`, but `True & True & True & True = True`

Reply:
No. `success` is changed (i.e., set to `False`) only when a file fails, and it is never reset otherwise. This means that at the end, it indicates if at least one file fails.